runtime/lua/vim/filetype.lua

```diff
@@ -863,6 +898,184 @@ local extension = {
+      return "edif"
+    end
+    return "clojure"
+  end,
+  smil = function(path, bufnr)
+    if getline(bufnr, 1):find("<?%s*xml.*?>") then
+      return "xml"
+    end
+    return "smil"
+  end,
+  smi = function(path, bufnr)
+    if getline(bufnr, 1):find("<smil>") then
+      return "smil"
+    end
+    return "mib"
+  end,
+  install = function(path, bufnr)
+    if getline(bufnr, 1):find("<%?php") then
+      return "php"
+    end
+    vim.fn["dist#ft#SetFileTypeSH"]("bash")
+  end,
+  pm = function(path, bufnr)
+    if getline(bufnr, 1):find("XPM2") then
+      return "xpm2"
+    elseif getline(bufnr, 1):find("XPM") then
+      return "xpm"
+    else
+      return "perl"
+    end
+  end,
+  me = function(path, bufnr)
+    local filename = vim.fn.fnamemodify(path, ":t")
+    if filename ~= "read.me" and filename ~= "click.me" then
+      return "nroff"
+    end
+  end,
+  reg = function(path, bufnr)
+    local line = getline(bufnr, 1)
+    if line:find("^REGEDIT[0-9]*%s*$") or line:find("^Windows Registry Editor Version %d*%.%d*%s*$") then
+      return "registry"
+    end
+  end,
+  decl = function(path, bufnr)
+    if getline(bufnr, 1, 3):lower():find("^<!sgml") then
+      return "sgmldecl"
+    end
+  end,
+  dcl = function(path, bufnr)
+    if getline(bufnr, 1, 3):lower():find("^<!sgml") then
+      return "sgmldecl"
+    end
+    return "clean"
+  end,
+  dec = function(path, bufnr)
+    if getline(bufnr, 1, 3):lower():find("^<!sgml") then
+      return "sgmldecl"
+    end
+  end,
+  web = function(path, bufnr)
+    if getline(bufnr, 1, 5):find("^%%") then
+      return "web"
+    end
+    return "winbatch"
+  end,
+  ttl = function(path, bufnr)
+    local line = getline(bufnr, 1)
+    if line:find("^@?prefix") or line:find("^@?base") then
+      return "turtle"
+    end
+    return "teraterm"
+  end,
+  am = function(path, bufnr)
+    local filename = vim.fn.fnamemodify(path, ":t"):lower()
+    if not filename:find("makefile.am$") then
+      return "elf"
+    end
+  end,
+  ["m4"] = function(path, bufnr)
+    local filename = vim.fn.fnamemodify(path, ":t"):lower()
+    if not filename:find("html.m4$") and not filename:find("fvwm2rc") then
+      return "m4"
+    end
+  end,
+  hw = function(path, bufnr)
+    if getline(bufnr, 1):find("<%?php") then
+      return "php"
+    end
+    return "virata"
+  end,
+  module = function(path, bufnr)
+    if getline(bufnr, 1):find("<%?php") then
+      return "php"
+    end
+    return "virata"
+  end,
+  pkg = function(path, bufnr)
+    if getline(bufnr, 1):find("<%?php") then
+      return "php"
+    end
+    return "virata"
+  end,
+  ms = function(path, bufnr)
+    if vim.fn["dist#ft#FTnroff"]() == 0 then
+      return "xmath"
+    end
+  end,
+  t = function(path, bufnr)
+    if vim.fn["dist#ft#FTnroff"]() == 0 and vim.fn["dist#ft#FTperl"]() == 0 then
+      return "tads"
+    end
+  end,
+  class = function(path, bufnr)
+    if not getline(bufnr, 1):find("^\xca\xfe\xba\xbe") then
```

Comment:
`\x` matches a hex digit. I would change this to:
```lua
local hex = "[0-9A-Fa-f]"
if not getline(bufnr, 1):find("^" .. hex .. "ca" .. hex .. "fe" .. hex .. "ba" .. hex .. "be")
```

Reply:
Lua patterns can use `%x` for this purpose.

Reply:
Didn't know that, that's better :+1: